go

```diff
@@ -88,6 +88,34 @@ def ci_build
   build
 end
 
+AUTOMATED_DEPLOY_PULL_CMD = "git pull && git submodule update --remote"
+INTERNAL_BUNDLE_CMD = "/usr/local/rbenv/shims/bundle"
+PUBLIC_BUNDLE_CMD = "/opt/install/rbenv/shims/bundle"
+JEKYLL_BUILD_CMD = "exec jekyll b"
+JEKYLL_PUBLIC_CONFIG = "--config _config.yml,_config_public.yml"
+
+def deploy_submodules
+  exec_cmd(AUTOMATED_DEPLOY_PULL_CMD + ' && cd _data && ' +
+    '/usr/local/rbenv/shims/ruby ./import-public.rb && cd .. && ' +
+    'git add _data/private _data/public/ pages/private && ' +
+    'git commit -m \'Private submodule update\' && git push')
+end
+
+def deploy_internal
+  exec_cmd("%s && %s && %s %s && %s %s %s" % [
+    AUTOMATED_DEPLOY_PULL_CMD,
+    INTERNAL_BUNDLE_CMD,
+    INTERNAL_BUNDLE_CMD, JEKYLL_BUILD_CMD,
+    INTERNAL_BUNDLE_CMD, JEKYLL_BUILD_CMD, JEKYLL_PUBLIC_CONFIG])
+end
```

Comment:
I don't use string replacement very often... the string to be replaced is confusing to look at, though I like how you laid out the arguments to make things more obvious.

I took a crack at refactoring, but thinking I made things worse.  Anyway, here it is... take it or leave it.

```ruby
def join_args(args)
  args.join(' ')
end

def join_commands(commands)
  commands.join(' && ')
end

def bundle_cmd(command=nil)
  args = [INTERNAL_BUNDLE_CMD, command].compact
  join_args(args)
end

def deploy_internal
  command = join_commands([
    AUTOMATED_DEPLOY_PULL_CMD,
    bundle_cmd,
    bundle_cmd(JEKYLL_BUILD_CMD),
    bundle_cmd(join_args([JEKYLL_BUILD_CMD, JEKYLL_PUBLIC_CONFIG]))
  ])
  exec_cmd(command)
end
```